src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle sides must be greater than zero');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      p * (p - this.a) * (p - this.b) * (p - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    const area: number = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    private width: number,
+    private height: number,
+  ) {
+    if (width <= 0) {
+      throw new Error('Width must be greater than 0');
```

Comment:
[CODE STYLE] - The error message should be more descriptive. Consider specifying that the error relates to the rectangle's width. For example: 'Rectangle width must be greater than 0 to form a valid rectangle.'